### PR TITLE
[Collect] fix some filter state inconsistencies

### PR DESCRIPTION
**Pull request by l2succes**
Fixes:
- https://artsyproduct.atlassian.net/browse/GROW-915

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 5ba7b3d37f7f65a2f88898ee8a4cf34219590ef8

https://deploy-preview-1300--artsy-reaction.netlify.com

File: src/Apps/Collect/FilterState.tsx [hunk 1]
```diff
@@ -111,89 +100,44 @@ export class FilterState extends Container<State> {
...
-        newPartialState[filter] = true
-      } else {
-        selectedFilters = without(selectedFilters, filter)
-        newPartialState[filter] = null
-      }
-    }
 
-    if (["major_periods", "partner_id", "medium"].includes(filter)) {
-      selectedFilters = selectedFilters.concat(["radio"])
+    switch (filter) {
```
Comment (xtina-starr) [inline src/Apps/Collect/FilterState.tsx]: 🙌

File: src/Apps/Collect/FilterState.tsx [hunk 2]
```diff
@@ -46,52 +39,48 @@ export class FilterState extends Container<State> {
...
+            break
+          default:
+            this.state[filter] = value
         }
       })
     }
   }
 
+  get filteredState() {
+    return omitBy(this.state, isNil)
```
[the remark this answers is not in the document]
Comment (l2succes) [reply src/Apps/Collect/FilterState.tsx]: Yup, any value that is "nullish" including `""` (https://lodash.com/docs/4.17.10#isNil)

**Comment by xtina-starr** [comment]
This looks legit to me. Using the `switch` statements makes this more clear. It might be worth adding a test for `unsetFilter`.

File: src/Apps/Collect/FilterState.tsx [hunk 3]
```diff
@@ -46,52 +39,48 @@ export class FilterState extends Container<State> {
...
-            this.state[filter] = props[filter] ? true : null
-          } else {
-            this.state[filter] = props[filter]
-          }
+        const value = props[filter]
+        if (!value) {
+          return
+        }
+
+        switch (filter) {
```
[the remark this answers is not in the document]
Comment (l2succes) [reply src/Apps/Collect/FilterState.tsx]: Yeah I think so :)

**Comment by xtina-starr** [comment]
Also I guess there is never really a need to unset the `page` property? I noticed that we explicitly address it in the `setFilter` method but not in `unsetFilter`.

**Comment by l2succes** [comment]
Right, after thinking it through some more, setting or unsetting any filter should reset the `page` property to `1` as the list would always change in theory so it's a special case.

**Comment by l2succes** [comment]
I'm gonna merge this one. I think we can still add a few more tests but this should cover our bases for the `FilterState` component

**Pull request merged**